metadata-ingestion/src/datahub/ingestion/transformer/base_transformer.py

```diff
@@ -180,6 +195,33 @@ def _transform_or_record_mcpw(
             self._record_mcp(envelope.record)
         return envelope if envelope.record.aspect is not None else None
 
+    def _handle_end_of_stream(
+        self, envelope: RecordEnvelope, entity_urn: str
+    ) -> Iterable[RecordEnvelope]:
+
+        if not isinstance(self, SingleAspectTransformer):
+            return
+
+        mcps: List[MetadataChangeProposalWrapper] = self.handle_end_of_stream(
+            entity_urn
+        )
+
+        for mcp in mcps:
+            # I am not sure if we need to update the work_unit_id. @harshal please confirm it
```

Comment:
no, we don't need to update the work_unit_id

Reply:
done